File: AAC-Practice/app/src/main/java/com/bowoon/android/aac_practice/BasicApp.java
package com.bowoon.android.aac_practice;

import android.app.Application;

public class BasicApp extends Application {
    @Override
    public void onCreate() {
        super.onCreate();
    }

    public AppDatabase getDatabase() {
        return AppDatabase.getInstance(this);
    }

    public DataRepository getRepository() {
        return DataRepository.getInstance(getDatabase());
    }
}

File: AACPracticeKotlin/app/src/main/java/com/bowoon/android/aac_practice_kotlin/BasicApp.kt
package com.bowoon.android.aac_practice_kotlin

import android.app.Application

class BasicApp : Application() {

    private fun getDatabase(): AppDatabase {
        return AppDatabase.Singleton.getInstance(this)!!
    }

    fun getRepository(): DataRepository {
        return DataRepository.Singleton.getInstance(getDatabase())
    }
}